x-pack/plugins/elastic_assistant/server/routes/attack_discovery/helpers.ts

```diff
@@ -71,3 +158,193 @@ export const getAssistantToolParams = ({
   request,
   size,
 });
+
+export const attackDiscoveryStatus: { [k: string]: AttackDiscoveryStatus } = {
+  failed: 'failed',
+  running: 'running',
+  succeeded: 'succeeded',
+};
+const MAX_GENERATION_INTERVALS = 5;
+
+export const addGenerationInterval = (
+  generationIntervals: GenerationInterval[],
+  generationInterval: GenerationInterval
+): GenerationInterval[] => {
+  const newGenerationIntervals = [generationInterval, ...generationIntervals];
+
+  if (newGenerationIntervals.length > MAX_GENERATION_INTERVALS) {
+    return newGenerationIntervals.slice(0, MAX_GENERATION_INTERVALS); // Return the first MAX_GENERATION_INTERVALS items
+  }
+
+  return newGenerationIntervals;
+};
+
+export const updateAttackDiscoveryStatusToRunning = async (
+  dataClient: AttackDiscoveryDataClient,
+  authenticatedUser: AuthenticatedUser,
+  apiConfig: ApiConfig
+): Promise<{
+  currentAd: AttackDiscoveryResponse;
+  attackDiscoveryId: string;
+}> => {
+  const foundAttackDiscovery = await dataClient?.findAttackDiscoveryByConnectorId({
+    connectorId: apiConfig.connectorId,
+    authenticatedUser,
+  });
+  let currentAd: AttackDiscoveryResponse;
+  let attackDiscoveryId: string;
+  if (foundAttackDiscovery == null) {
+    const ad = await dataClient?.createAttackDiscovery({
+      attackDiscoveryCreate: {
+        attackDiscoveries: [],
+        apiConfig,
+        status: attackDiscoveryStatus.running,
+      },
+      authenticatedUser,
+    });
+    if (ad == null) {
+      throw new Error(
+        `Could not create attack discovery for connectorId: ${apiConfig.connectorId}`
+      );
+    } else {
+      currentAd = ad;
+    }
+    attackDiscoveryId = currentAd.id;
+  } else {
+    attackDiscoveryId = foundAttackDiscovery.id;
+
+    const ad = await dataClient?.updateAttackDiscovery({
+      attackDiscoveryUpdateProps: {
+        id: attackDiscoveryId,
+        status: attackDiscoveryStatus.running,
+        backingIndex: foundAttackDiscovery.backingIndex,
+      },
+      authenticatedUser,
+    });
+    if (ad == null) {
+      throw new Error(
+        `Could not update attack discovery for connectorId: ${apiConfig.connectorId}`
+      );
+    } else {
+      currentAd = ad;
+    }
+  }
+  return {
+    currentAd,
+    attackDiscoveryId,
```

Comment:
To eliminate the `let` reassignment via `if` / `else` branching in the following:

```typescript
  let currentAd: AttackDiscoveryResponse;
  let attackDiscoveryId: string;
  if (foundAttackDiscovery == null) {
    const ad = await dataClient?.createAttackDiscovery({
      attackDiscoveryCreate: {
        attackDiscoveries: [],
        apiConfig,
        status: attackDiscoveryStatus.running,
      },
      authenticatedUser,
    });
    if (ad == null) {
      throw new Error(
        `Could not create attack discovery for connectorId: ${apiConfig.connectorId}`
      );
    } else {
      currentAd = ad;
    }
    attackDiscoveryId = currentAd.id;
  } else {
    attackDiscoveryId = foundAttackDiscovery.id;

    const ad = await dataClient?.updateAttackDiscovery({
      attackDiscoveryUpdateProps: {
        id: attackDiscoveryId,
        status: attackDiscoveryStatus.running,
        backingIndex: foundAttackDiscovery.backingIndex,
      },
      authenticatedUser,
    });
    if (ad == null) {
      throw new Error(
        `Could not update attack discovery for connectorId: ${apiConfig.connectorId}`
      );
    } else {
      currentAd = ad;
    }
  }
  return {
    currentAd,
    attackDiscoveryId,
```

consider something like (untested):

```typescript
  const attackDiscoveryId = foundAttackDiscovery?.id || '';
  const currentAd = foundAttackDiscovery
    ? await dataClient?.updateAttackDiscovery({
        attackDiscoveryUpdateProps: {
          backingIndex: foundAttackDiscovery.backingIndex,
          id: attackDiscoveryId,
          status: attackDiscoveryStatus.running,
        },
        authenticatedUser,
      })
    : await dataClient?.createAttackDiscovery({
        attackDiscoveryCreate: {
          apiConfig,
          attackDiscoveries: [],
          status: attackDiscoveryStatus.running,
        },
        authenticatedUser,
      });

  if (!currentAd) {
    throw new Error(
      `Could not ${foundAttackDiscovery ? 'update' : 'create'} attack discovery for connectorId: ${
        apiConfig.connectorId
      }`
    );
  }

  return {
    attackDiscoveryId: currentAd.id,
    currentAd,
  };
};
```

Reply:
https://github.com/elastic/kibana/pull/184949/commits/803d19adc32878808d32ec733846e2199dc1f236